sdks/python/apache_beam/runners/interactive/sql/beam_sql_magics.py

```diff
@@ -181,11 +196,22 @@ def beam_sql(self, line: str, cell: Optional[str] = None) -> Optional[PValue]:
     if not query:
       on_error('Please supply the SQL query to be executed.')
       return
+    if runner and runner not in _SUPPORTED_RUNNERS:
+      on_error(
+          'Runner "%s" is not supported. Supported runners are %s.',
+          runner,
+          _SUPPORTED_RUNNERS)
     query = ' '.join(query)
 
-    found = find_pcolls(query, pcoll_by_name(), verbose=verbose)
+    found = find_pcolls(
+        query,
+        pcoll_by_name(),
+        run=runner in ('DirectRunner', None),
+        verbose=verbose)
+    schemas = set()
+    main_session = importlib.import_module('__main__')
```

Comment:
Move this line to right before the hasattr(main_session), since it's not used elsewhere

Reply:
It's used below in a for-loop, put it here to only import it once.